### PR TITLE
PVP-80-PVP-81 User Information Update Functionality

### PR DESCRIPTION
**Pull request by AurisTFG**
Added functionality in the profile view to let the user update their username, mass and height.

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -80,25 +146,116 @@ private fun ProfileFooter(
...
+    label: String,
+    value: String,
+    dialogTitle: @Composable () -> Unit,
+    dialogContent: @Composable () -> Unit,
+    onConfirmClick: () -> Unit
+) {
+    Box(
+        modifier = Modifier
+            .background(
+                color = Color.LightGray,
```
Comment (vytjar) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:217]: Should not use fixed colors. Use MaterialTheme.colorScheme.* . This way the colors change based on dark mode/light mode settings.

**Review by vytjar — CHANGES_REQUESTED** [review]
Also use CTRL + ALT + L on the code to save Klaidas from having to write 30 comments (like when reviewing my code)

File: app/src/main/java/com/pvp/app/ui/common/Modifiers.kt [hunk 2]
```diff
@@ -87,4 +90,16 @@ fun Modifier.fadingEdge(brush: Brush) = this
...
+fun Modifier.underline(): Modifier = this
+    .drawBehind {
+        val verticalOffset = size.height - 2.sp.toPx()
+
+        drawLine(
+            color = Color.Black,
+            strokeWidth = 1.dp.toPx(),
+            start = Offset(0f, verticalOffset),
+            end = Offset(size.width, verticalOffset)
+        )
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/common/Modifiers.kt:104]: Multiple parameters _could_ be split.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -2,54 +2,120 @@ package com.pvp.app.ui.screen.profile
...
+                        keyboardType = KeyboardType.Number
+                    ),
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(bottom = 16.dp)
+                )
+            },
+            onConfirmClick = {
+                onUpdateHeight(heightEditing)
+                heightDisplay = heightEditing
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:117]: Single empty line needed between method/assignation.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -80,25 +146,116 @@ private fun ProfileFooter(
...
+                        onValueChange = { userNameEditing = it },
+                        label = { Text("Username") },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(bottom = 16.dp)
+                    )
+                },
+                onConfirmClick = {
+                    onUpdateUsername(userNameEditing)
+                    userNameDisplay = userNameEditing
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:193]: Same as above.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt (hunk 1 above)
Comment (klaidoshka) [reply app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:217]: Follow throughout the whole file.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 5]
```diff
@@ -80,25 +146,116 @@ private fun ProfileFooter(
...
+    Box(
+        modifier = Modifier
+            .background(
+                color = Color.LightGray,
+                shape = RoundedCornerShape(8.dp))
+            .padding(8.dp)
+            .fillMaxWidth()
+    ) {
+        Row(
+            modifier = Modifier.fillMaxWidth(), // Apply fillMaxWidth() to Row
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:223]: Remove redundant comments.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 6]
```diff
@@ -144,3 +311,59 @@ fun ProfileScreen(
...
+                title = dialogTitle,
+                text = dialogContent,
+                onDismissRequest = {
+                    showDialog.value = false
+                },
+                confirmButton = {
+                    Button(
+                        onClick = {
+                            showDialog.value = false
+                            onConfirmClick()
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:351]: Separate different code blocks.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt [hunk 7]
```diff
@@ -30,6 +30,36 @@ class ProfileViewModel @Inject constructor(
         }
     }
 
+    fun updateUsername(newUsername: String) {
+        viewModelScope.launch {
+            _user.value?.username = newUsername
+
+            _user.value?.let {
+                userService.merge(it)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt:38]: Could be refactored into
```kotlin
_user.value?.let {
  it.username = newUsername

  userService.merge(it)
}
```

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt [hunk 8]
```diff
@@ -30,6 +30,36 @@ class ProfileViewModel @Inject constructor(
...
+            }
+        }
+    }
+
+    fun updateMass(newMass: Int) {
+        viewModelScope.launch {
+            _user.value?.mass = newMass
+
+            _user.value?.let {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt:47]: Same as above.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt [hunk 9]
```diff
@@ -30,6 +30,36 @@ class ProfileViewModel @Inject constructor(
...
+                userService.merge(it)
+            }
+        }
+    }
+
+    fun updateHeight(newHeight: Int) {
+        viewModelScope.launch {
+            _user.value?.height = newHeight
+
+            _user.value?.let {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt:57]: Same as above.

**Review by klaidoshka — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 10]
```diff
@@ -2,54 +2,131 @@ package com.pvp.app.ui.screen.profile
...
+            onConfirmClick = {
+                val newHeight = heightEditing.toIntOrNull() ?: 0
 
-        Text(
-            text = "Mass: ${user.value?.mass}",
-            style = MaterialTheme.typography.bodyMedium
+                if (newHeight in 40..300) {
+                    onUpdateHeight(newHeight)
+                    heightDisplay = newHeight
+                    context.showToast(message = "Your height has been updated!")
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:125]: Different blocks of code could be separated.

Method,
Assignation,
Method.

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 11]
```diff
@@ -2,54 +2,131 @@ package com.pvp.app.ui.screen.profile
...
+            onConfirmClick = {
+                val newMass = massEditing.toIntOrNull() ?: 0
 
-        Text(
-            text = "Points: ${user.value?.points}",
-            style = MaterialTheme.typography.bodyMedium
+                if (newMass in 2..700) {
+                    onUpdateMass(newMass)
+                    massDisplay = newMass
+                    context.showToast(message = "Your mass has been updated!")
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:95]: Different blocks of code could be separated.

Method,
Assignation,
Method.

**Review by klaidoshka — APPROVED** [review]

**Review by vytjar — APPROVED** [review]

**Pull request merged**